apps/api/src/modules/webhook/webhook.controller.ts

```diff
@@ -0,0 +1,21 @@
+import { Controller, Post, Body, Param, HttpException, HttpStatus } from '@nestjs/common';
+import { WebhookService } from './webhook.service';
+
+@Controller('providers')
+export class WebhookController {
+  constructor(private readonly webhookService: WebhookService) {}
+
+  @Post(':providerId/webhooks')
+  async registerWebhook(
+    @Param('providerId') providerId: number,
+    @Body('webhookUrl') webhookUrl: string,
+  ): Promise<{ status: string; message: string }> {
+    try {
+      await this.webhookService.registerWebhook(providerId, webhookUrl);
+      return { status: 'success', message: 'Webhook registered successfully.' };
+    } catch (error) {
+      console.error('Error registering webhook:', error);
```

Comment:
Run linting console logs should be triggered as error. if its not added in eslint config add it.

Reply:
the configuration is added 
    'no-console': 'warn',
  i tried added console.error again now it logging it as warning don't know why it didn't get detected earlier , i also try adding in the catch blocks

Reply:
Linting should not be passed for such case. Can add it as a Task to check why it was not triggered, as we want to keep warnings as 0 too in CI.